File: Data Structures & Algorithms/is-subsequence/submission-1.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:

        def dfs(i, j):
            if i >= len(s):
                return True
            
            if j >= len(t):
                return False
            
            pass_one = False
            if s[i] == t[j]:
                pass_one = dfs(i + 1, j + 1)
            
            if pass_one:
                return pass_one
            else:
                return dfs(i, j + 1)
        
        return dfs(0, 0)

            

File: Data Structures & Algorithms/subsets-ii/submission-1.java
class Solution {
    private void dfs(int[] nums, int start, List<Integer> subset, List<List<Integer>> subsets) {
        subsets.add(new ArrayList<>(subset));

        for (int i = start; i < nums.length; i++) {

            if (i > start && nums[i - 1] == nums[i]) {
                continue;
            }

            subset.add(nums[i]);
            dfs(nums, i + 1, subset, subsets);
            subset.remove(subset.size() - 1);
        }
    }
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> subsets = new ArrayList<>();
        dfs(nums, 0, new ArrayList<>(), subsets);
        return subsets;
    }
}
